pkg/git/proxy.go

```diff
@@ -0,0 +1,30 @@
+package git
+
+import (
+	git2go "github.com/libgit2/git2go/v31"
+	giturls "github.com/whilp/git-urls"
+	"golang.org/x/net/http/httpproxy"
+)
+
+func proxyFromEnv(gitURL string) (git2go.ProxyOptions, error) {
+	u, err := giturls.Parse(gitURL)
```

Comment:
Should we use `gitUrls` [here](https://github.com/pivotal/kpack/blob/ccb2f75404c088b8c2c9f9f585f1e1ec2c7ffa57/pkg/git/git_keychain.go#L180)?

Reply:
nice!